src/wire/ipv4.rs

```diff
@@ -110,6 +110,19 @@ impl Cidr {
         Cidr { address, prefix_len }
     }
 
+    /// Create an IPv4 CIDR block from the given address and network mask.
+    ///
+    /// # Panics
+    /// This function panics if the prefix length is larger than 32.
+    pub fn with_netmask(addr: Address, netmask: Address) -> Cidr {
+        let prefix_len = netmask.0[0].count_ones()
+                         + netmask.0[1].count_ones() 
+                         + netmask.0[2].count_ones()
+                         + netmask.0[3].count_ones();
```

Comment:
I think this function must check that the ones in netmask are all consecutive and left-aligned, and return a `Result`.

Reply:
wow, you are very professional.